### PR TITLE
DOC: first attempt at documenting load_library.

### PR DESCRIPTION
**Pull request by pizzathief**
Found some doco at
http://docs.scipy.org/doc/numpy-dev/user/c-info.python-as-glue.html
which seemed to describe the function, so used that.

### CODE REVIEW

File: numpy/ctypeslib.py [hunk 1]
```diff
@@ -89,6 +89,33 @@ def _dummy(*args, **kwds):
...
+        It is possible to load a library using 
+        >>> lib = ctypes.cdll[<full_path_name>]
+
+        But there are cross-platform considerations, such as library file extensions,
+        plus the fact Windows will just load the first library it finds with that name.  
+        Numpy supplies the load_library function as a convenience.
+
+        Parameters
+        ----------
+        libname : string
```
Comment (charris) [inline numpy/ctypeslib.py]: `str` instead of `string`

File: numpy/ctypeslib.py [hunk 2]
```diff
@@ -89,6 +89,33 @@ def _dummy(*args, **kwds):
...
+        Returns
+        -------
+        ctypes.cdll[libpath] : library object
+           A ctypes library object 
+
+        Raises
+        ------
+        OSError
+            If there is no library with the expected extension, or the 
+            library is defective and can't be loaded.
```
Comment (charris) [inline numpy/ctypeslib.py]: `cannot` is a bit safer than `can't`, although that does not matter here.

**Comment by charris** [comment]
LGTM modulo a couple of nitpicks.

**Comment by charris** [comment]
@pizzathief Be good to finish this up.

**Comment by rgommers** [comment]
LGTM too, merging. Thanks @pizzathief

**Pull request merged**